### PR TITLE
added parent, siblings, children part methods.

### PR DESCRIPTION
**Pull request by jberends**
merged with pull request #37, see issue #35 

This is a continuation wil additional testing.

also created the following `Part` methods:
* parent()
* children()
* siblings() (== parent().children())

### CODE REVIEW

File: pykechain/models/part.py [hunk 1]
```diff
@@ -11,16 +11,30 @@ def __init__(self, json, **kwargs):
         super(Part, self).__init__(json, **kwargs)
 
         self.category = json.get('category')
+        if 'parent' in json and json.get('parent') and 'id' in json.get('parent'):
```
Comment (wilcoschoneveld) [inline pykechain/models/part.py]: `self.parent_id = json['parent'].get('id') if 'parent' in json else None`

File: pykechain/models/partset.py [hunk 2]
```diff
@@ -11,7 +11,10 @@ class PartSet(object):
 
     def __init__(self, parts):
         """Construct a PartSet from a part iterable."""
-        self._parts = list(parts)
+        if parts:
+            self._parts = list(parts)
```
Comment (wilcoschoneveld) [inline pykechain/models/partset.py]: I think parts should always be an iterable here

File: pykechain/models/part.py (hunk 1 above)
Comment (jberends) [inline pykechain/models/part.py]: @wilcoschoneveld, if the json['parent'] exists in json BUT is 'None' we cannot query it for a subsequent dictionary. That is why there is an extra `json.get('parent') is not None` in the check.

    self.parent_id = json['parent'].get('id') if 'parent' in json and json.get('parent') else None

this works and all test pass in 35, will push and see if travis plays nice as well (in 27?). Nice compact code

File: pykechain/models/partset.py (hunk 2 above)
Comment (jberends) [inline pykechain/models/partset.py]: i check the case that parts are provided as None. If the list of parts are provided as [], the self._parts = list([]), which is a list with no content.

**Pull request merged**